RecoMuon/MuonIdentification/python/thinnedTrackExtras_cff.py

```diff
@@ -0,0 +1,65 @@
+from RecoMuon.MuonIdentification.muonTrackExtraThinningProducer_cfi import muonTrackExtraThinningProducer
+from RecoTracker.TrackProducer.trackingRecHitThinningProducer_cfi import trackingRecHitThinningProducer
+from RecoTracker.TrackProducer.siPixelClusterThinningProducer_cfi import siPixelClusterThinningProducer
+from RecoTracker.TrackProducer.siStripClusterThinningProducer_cfi import siStripClusterThinningProducer
+
+import FWCore.ParameterSet.Config as cms
+
+thinnedGeneralTrackExtras = muonTrackExtraThinningProducer.clone(inputTag = cms.InputTag("generalTracks"),
+                                                                  cut = cms.string("pt > 3. || isPFMuon"),
```

Comment:
The two PRs are mutually exclusive. Let me however put a note here that the corresponding cut in #31217 is
```
 "pt > 5 || (pt > 4.5 && (isGlobalMuon || isStandAloneMuon || numberOfMatches > 0 || muonID(\"RPCMuLoose\")))"
```
see https://github.com/cms-sw/cmssw/pull/31217/files#diff-25b6ffc7f1a7fb0ddc2a61dd171b1a06R48

Reply:
Right, I will update the cuts in this PR as part of the rebase on top of https://github.com/cms-sw/cmssw/pull/31100

The logic will still be slightly different however, since the thinning can proceed in two stages.

So the cut will be ```pt > 4.5``` at reconstruction level, and then again ```pt > 4.5``` at the MINIAOD step (but implicitly inheriting the additional requirements for muons below 5GeV as it acts on slimmedMuons in this PR, but on the full muons collection in https://github.com/cms-sw/cmssw/pull/31217)